scripts/commands/checkLinks.ts

```diff
@@ -59,104 +58,21 @@ const readArgs = (): Arguments => {
     .parseSync();
 };
 
-/**
- * Process the markdown and Jupyter notebook files.
- *
- * filePathsToLoad is for files that we only need to load
- * so that we can determine if links from other files to that
- * file are valid. We need to determine all of its anchors.
- *
- * filePathsToCheck is for files that we will actually end up
- * checking. We determine its internal and external links. Other
- * files might link to these files, too, so we also determine
- * the valid anchors to make sure those links are correct.
- *
- * Returns a triplet:
- *   1. A list of `File` objects with their anchors. These represent
- *      the universe of valid internal links, other than any additional
- *      we may add in main() for e.g. images.
- *   2. A list of Link objects with internal links we will validate.
- *   3. A list of Link objects with external links we will validate.
- */
-async function processDocsFiles(
-  filePathsToLoad: string[],
-  filePathsToCheck: string[],
-): Promise<[File[], Link[], Link[]]> {
-  const files: File[] = [];
-  for (let filePath of filePathsToLoad) {
-    const [_, anchors] = await getMarkdownAndAnchors(filePath);
-    files.push(new File(filePath, anchors, false));
-  }
-
-  const linksToOriginFiles = new Map<string, string[]>();
-  for (const filePath of filePathsToCheck) {
-    const [markdown, anchors] = await getMarkdownAndAnchors(filePath);
-    files.push(new File(filePath, anchors, false));
-    await addLinksToMap(filePath, markdown, linksToOriginFiles);
-  }
-
-  const internalLinks: Link[] = [];
-  const externalLinks: Link[] = [];
-  for (let [linkPath, originFiles] of linksToOriginFiles) {
-    originFiles = originFiles.filter(
-      (originFile) =>
-        FILES_TO_IGNORES[originFile] == null ||
-        !FILES_TO_IGNORES[originFile].includes(linkPath),
-    );
-
-    if (originFiles.length > 0) {
-      const link = new Link(linkPath, originFiles);
-      link.isExternal ? externalLinks.push(link) : internalLinks.push(link);
-    }
-  }
-
-  return [files, internalLinks, externalLinks];
-}
-
 async function main() {
   const args = readArgs();
 
-  const docsToLoad = await globby(DOCS_GLOBS_TO_LOAD);
-  const docsToCheck = await globby(DOCS_GLOBS_TO_CHECK);
-
-  // Parse the files with links and get a list with all the links
-  // in all the files without duplications.
-  const [docsFiles, internalLinkList, externalLinkList] =
-    await processDocsFiles(docsToLoad, docsToCheck);
-
-  // Define extra files that we don't want to parse
+  const fileBatch = await FileBatch.fromGlobs(
+    DOCS_GLOBS_TO_CHECK,
+    DOCS_GLOBS_TO_LOAD,
+  );
   const otherFiles = [
     ...(await globby("{public,docs}/**/*.{png,jpg,gif,svg}")).map(
-      (fp) => new File(fp, [], false),
+      (fp) => new File(fp, []),
     ),
     ...SYNTHETIC_FILES.map((fp) => new File(fp, [], true)),
   ];
```

Comment:
Just as an idea. What do you think about having the `otherFiles` in the constructor of the `fileBatch` class and using the `fromGlobs` for them? By doing that, we can avoid having an extra argument in the check call. I think that the argument can be confusing because we already provided files to the class.

Reply:
Yeah, I think that could work too. The reason I didn't do it is we'd end up recomputing `otherFiles` many times in the follow up PR to add historical APIs, since each historical version is its own `FileBatch`.

Reply:
Oh, I see, that's a good reason. However, this list loads all the images for all the versions every time, when perhaps it's not necessary to load any image besides the ones belonging to that version, and maybe the current one. For example, Qiskit v0.33 doesn't need to have the images in `public/images/api/qiskit/0.19/` load. Regardless of that, I like a lot you did the refactor, so if you think it's okay because the amount of images is not that large, I'm good with it too :smiley: